yjit/src/codegen.rs

```diff
@@ -5754,23 +5754,118 @@ fn jit_rb_str_empty_p(
     return true;
 }
 
-// Codegen for rb_str_concat() -- *not* String#concat
-// Frequently strings are concatenated using "out_str << next_str".
-// This is common in Erb and similar templating languages.
-fn jit_rb_str_concat(
+// Codegen for rb_str_concat() with an integer argument -- *not* String#concat
+// Using strings as a byte buffer often includes appending byte values to the end of the string.
+fn jit_rb_str_concat_codepoint(
     jit: &mut JITState,
     asm: &mut Assembler,
     _ci: *const rb_callinfo,
     _cme: *const rb_callable_method_entry_t,
-    _block: Option<BlockHandler>,
+    _block: Option<crate::codegen::BlockHandler>,
```

Comment:
nitpick but why do you need to specify `crate::codegen::` here? Isn't that the module we're currently in?

Reply:
I think the IDE inserted that. I'll fix it. Thanks for catching it.